packages/cli/src/lifecycles/config.js

```diff
@@ -125,13 +129,13 @@ const readAndMergeConfig = async () => {
     // workspace validation
     if (workspace) {
       if (!(workspace instanceof URL)) {
-        return Promise.reject("Configuration error: workspace must be an instance of URL");
+        throw new Error("Configuration error: workspace must be an instance of URL");
       }
 
       if (await checkResourceExists(workspace)) {
         customConfig.workspace = workspace;
       } else {
-        return Promise.reject(
+        throw new Error(
```

Comment:
same as my comment above, let's avoid any additional refactorings in this PR not directly related to fixing all the loops and things just to try and keep the diff as small and focused as possible .  (we can always just open a separate PR for this if we feel strongly about it)